utils/TimeoutManager.ts

```diff
@@ -0,0 +1,14 @@
+export class TimeoutManager {
+  timeout: NodeJS.Timeout | undefined
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  execute(callback: (...args: any[]) => any, delay: number) {
+    this.timeout = setTimeout(callback, delay)
+  }
+
+  cancel() {
+    this.timeout && clearTimeout(this.timeout)
+  }
+}
+
+export default TimeoutManager
```

Comment:
I think this would be more idiomatic as a custom hook?

```ts
const useTimeout = () => {
  const [$id, $setId] = useState<NodeJS.Timeout | undefined>(undefined)
  return {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    execute: useCallback(
      (fn: (any[]) => any, delay: number) => $setId(setTimeout(fn, delay)),
      [$setId]
    ),
    cancel: useCallback(
      () => clearTimeout($id),
      [$id]
    )
  } 
}
```

Then downstream:

```ts
const {execute, cancel} = useTimer()
```

?

Reply:
I think no. Any updating of state will call rerender of component where it will be used.

Reply:
I'm not convinced this is a good pattern but in favor of getting this in I'm fine with this for now.